### PR TITLE
Update MVC lesson example, add MDN link to MVC

### PR DESCRIPTION
**Pull request by sertmer**
This PR addressed issue #710 

## Summary of changes needed: 
- [x] The current lesson uses a class based component to describe the MVC pattern
- [x] This example needs to be updated to be a functional component
- [x] Any language in the lesson that is referring to that class based component may also need to be updated
- [x] The lesson should include a link to the MVC pattern architecture page on MDN

### Changes Made:

- [x] All of the above

### Questions
- Please review my react code - I was feeling a bit rusty!
- No other specific considerations

### CODE REVIEW

File: lessons/module-3/mvc.md [hunk 1]
```diff
@@ -63,35 +63,31 @@ for this pattern, you will find it all over the place.
...
 
-class App extends Component {
-  constructor() {
-    super()
-    this.state = {
-      count: 0,
-    }
-  }
+function App() {
+  const [count, setCount] = useSate(0);
```
Comment (hfaerber) [inline lessons/module-3/mvc.md]: Just missing a 't' in state here :)

File: lessons/module-3/mvc.md [hunk 2]
```diff
@@ -63,35 +63,31 @@ for this pattern, you will find it all over the place.
...
 "Data flow in MVC")
 
+Take a look at [the docs on MDN](https://developer.mozilla.org/en-US/docs/Glossary/MVC) if you'd like to see it explained in another way.
+
 ### Ok, by why would I ever use it?
 
 You've already been using it! Let's consider a super simple example:
 
 ```js
 import React, { Component } from 'react';
```
Comment (hfaerber) [inline lessons/module-3/mvc.md]: We should be able to remove the {Component} import here

**Review by hfaerber — CHANGES_REQUESTED** [review]
Thanks for working on this Scott!  I made 2 comments on the code example.
Could you also edit the very end of the lesson please?
- In the **Wrapping it up** section, can you please remove "How does Redux play into this? Work on this for 10 minutes, then we’ll reconvene."
- In the **References** section, can you please stack the two links so its clear there are two?

File: lessons/module-3/mvc.md (hunk 2 above)
Comment (sertmer) [reply lessons/module-3/mvc.md]: ✅

File: lessons/module-3/mvc.md (hunk 1 above)
Comment (sertmer) [reply lessons/module-3/mvc.md]: Thank you! ✅

**Comment by sertmer** [comment]
Thanks Heather! I made those changes and I'll merge it now!

**Pull request merged**